src/closedai/pipelines.py

```diff
@@ -0,0 +1,66 @@
+# Helpful links:
+# https://huggingface.co/docs/transformers/main/en/main_classes/pipelines#transformers.TextGenerationPipeline
+
+
+import time
+from threading import Thread
+
+from transformers import TextIteratorStreamer, pipeline
+
+
+class ClosedAIPipeline:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def generate_completion(self, text, **kwargs):
+        pass
+
+    def get_completion(self, text, **kwargs):
+        return "".join(list(self.generate_completion(text, **kwargs)))
+
+
+class DummyPipeline(ClosedAIPipeline):
+    def __init__(self, duration=5):
+        self.duration = duration
+
+    def generate_completion(self, text, **kwargs):
+        for i in range(1, self.duration + 1):
+            yield f", and {i}"
+            time.sleep(0.5)
+
+    def generate_chat_completion(self, messages, **kwargs):
+        for i in range(1, self.duration + 1):
+            yield f", and {i}"
+            time.sleep(0.5)
+
+
+class HuggingFacePipeline(ClosedAIPipeline):
+    def __init__(self, pipe=None, model=None, tokenizer=None, streamer=None, decode_kwargs=None):
+        if pipe is None and model is None and tokenizer is None:
```

Comment:
logic here is really dumb wrt the input args. sorry bout that. will make it cleaner/clearer later. for now use it by just instantiating this class w/ the default `pipe`/`model`/`tokenizer`/`streamer` of `None`.

Reply:
also its using brand new streamer classes, which probably are unstable API wise and only on main branch rn.

I think this is a great example of why folks probably will want to define their own w/ their own environment via some template docker app we help them create